guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,63 @@
+# Motivation
+Our changelogs (reviewed by users/customers) are autogenerated from PR titles and git commit text.  The better this information is, the more we help our customers.  This information also helps:
+
+* reviewers understand changes
+* current+future team members understand the history of changes
+* patch managers understand risk/importance
+* when hunting for regressions, helps us better understand what is/isn’t likely to be a cause.
+
+# PR Metadata
+
+__PR title:__ Bug NNNN: A short sentence with capitalization and no period or the Jira tag for the issue “Feature TEAM-XXX:”
+
+* Try to describe the problem the user sees, rather than the code change you are doing
+  * I.e. “Console does not work when user clicks X” instead of “Fix drawer notification integration with underlying controller”
+* Prefix bug fixes with the bug number, “Bug XXXXX:” 
+  * (capitalize bug, space, number, then colon)
+* Do not let your PR title get truncated/wrapped into the description
+* Additional prefixes like what release the PR applies to are allowed, but they will be stripped from the changelog (the changelog for 4.4 doesn’t need to say “release-4.4” 
+  * Changelog strips anything between square brackets that has no spaces - i.e. [release-4.4] is stripped but [release 4.4] would not be
+    
+
+__PR description:__  Additional context reviewers need to understand your PR
+
+* Why was this change needed?
+* What changes are being made?  
+* Reference to any prior work(other PRs) or follow up work that will be needed.
+
+[Canonical PR text example](https://github.com/openshift/openshift-apiserver/pull/69)
+
+# Git Commit Best practices
+
+## Number of Commits
+
+Minimize your number of commits per PR with a goal of a single commit in many cases.
```

Comment:
is it really the best indicator of better reviewable PR? if there are various commits with smaller logical changes, for such a PR is much easy to review instead of one commit with very large change.

personally I feel asking people to make one commit PRs would force people to put even large changes in one commit. 

Maybe instead of number of commit we talk about _non-vendor/non-generated_ changeset size?

Reply:
> is it really the best indicator of better reviewable PR? 

It was not claimed as the best indicator of anything. it was positioned as a desirable attribute within the conditions/caveats discussed later.

> if there are various commits with smaller logical changes, for such a PR is much easy to review instead of one commit with very large change.

there was discussion about this elsewhere in this PR, some people review commit by commit, others do it at a PR level.  Which approach makes sense depends in part of how the PR is structured.

> personally I feel asking people to make one commit PRs would force people to put even large changes in one commit.

I want to emphasize this as much as i can:  there is no requirement to make single commit PRs.  The ask is to think carefully about how many commits you are putting in a PR and why.  Single commit is a goal(where applicable) and not a requirement.

Reply:
>  Single commit is a goal(where applicable)

This is the concern, single commit is strongly recommended, so why bother with splitting for ease of maintainance/review/...?

Reply:
> This is the concern, single commit is strongly recommended, so why bother with splitting for ease of maintainance/review/...?

i'm going to reframe this statement to try to alleviate the concerns people are raising about it.